lib/auth/auth_test.go

```diff
@@ -1085,6 +1085,28 @@ func TestU2FSignChallengeCompat(t *testing.T) {
 	})
 }
 
+func TestEmitSSOLoginFailureEvent(t *testing.T) {
+	mockE := &events.MockEmitter{}
+
+	emitSSOLoginFailureEvent(context.Background(), mockE, "test", trace.BadParameter("some error"))
+	event := mockE.LastEvent().(*events.UserLogin)
```

Comment:
Instead of type assertion (which might potentially trigger a panic), assert the equivalence with `require`, e.g. it should suffice to:
```go
require.Equal(t, mockE.LastEvent(), &events.UserLogin{
	Metadata: events.Metadata{
		Type: events.UserLoginEvent,
		Code: events.UserSSOLoginFailureCode,
	},
	Method: "test",
	Status: events.Status{
		Success:     false,
		Error:       "some error",
		UserMessage: "some error",
	},
})
```
and matching on `Method` will not be necessary since it's also part of the expectation.

Reply:
revised